app/src/test/java/com/android/sample/model/image/ImageRepositoryFirestoreTest.kt

```diff
@@ -0,0 +1,207 @@
+  private lateinit var imageRepository: ImageRepositoryFirestore
+  @Mock private lateinit var mockDocumentSnapshot: DocumentSnapshot
+  @Mock private lateinit var mockDocumentReference: DocumentReference
+  @Mock private lateinit var mockCollectionReference: CollectionReference
+  @Mock private lateinit var mockListResult: ListResult
+  @Mock private lateinit var bitmapHelper: BitmapHelper
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.openMocks(this)
+    `when`(mockStorage.reference).thenReturn(mockStorageRef)
+    `when`(mockFirestore.collection("users")).thenReturn(mockCollectionReference)
+    `when`(mockCollectionReference.document(anyString())).thenReturn(mockDocumentReference)
+    imageRepository = ImageRepositoryFirestore(mockFirestore, mockStorage)
+  }
+
+  @Test
+  fun uploadProfilePicture_success() {
+    val userId = "userId"
+    val bitmap = mock(Bitmap::class.java)
+    val uri = Uri.parse("https://example.com/profile.jpg")
+    val uriString = "https://example.com/profile.jpg"
+    val baos = ByteArrayOutputStream()
+    bitmap.compress(Bitmap.CompressFormat.JPEG, 50, baos)
+    val imageData = baos.toByteArray()
+
+    `when`(mockStorageRef.child("users/$userId/profile_picture.jpg")).thenReturn(mockStorageRef)
+    `when`(mockStorageRef.putBytes(imageData)).thenReturn(mockUploadTask)
+
+    `when`(mockUploadTask.addOnSuccessListener(any())).thenAnswer { invocation ->
+      val listener = invocation.getArgument<OnSuccessListener<UploadTask.TaskSnapshot>>(0)
+      val taskSnapshot = mock(UploadTask.TaskSnapshot::class.java) // Mock the snapshot
+      listener.onSuccess(taskSnapshot)
+      mockUploadTask // continue the chain
+    }
+    `when`(mockUploadTask.addOnSuccessListener(any())).thenAnswer { invocation ->
+      val listener = invocation.getArgument<OnSuccessListener<UploadTask.TaskSnapshot>>(0)
+      val taskSnapshot = mock(UploadTask.TaskSnapshot::class.java) // Mock the snapshot
+      listener.onSuccess(taskSnapshot)
+      mockUploadTask // continue the chain
+    }
+
+    `when`(mockStorageRef.downloadUrl).thenReturn(Tasks.forResult(uri))
+    `when`(mockDocumentReference.update("photo", uriString)).thenReturn(Tasks.forResult(null))
+
+    var resultUri = ""
+    imageRepository.uploadProfilePicture(
+        userId,
+        bitmap,
+        {
+          resultUri = it
+          assertEquals(uriString, resultUri)
+        },
+        {})
+  }
+
+  @Test
+  fun fetchProfileImageUrl_success() {
+    val userId = "testUserId"
+    val expectedUri = "https://example.com/profile.jpg"
+
+    `when`(mockStorageRef.child("users/$userId/profile_picture.jpg")).thenReturn(mockStorageRef)
+    `when`(mockStorageRef.downloadUrl).thenReturn(Tasks.forResult(Uri.parse(expectedUri)))
+
+    var resultUrl = ""
+    imageRepository.fetchProfileImageUrl(
+        userId,
+        {
+          resultUrl = it
+          assertEquals(expectedUri, resultUrl)
+        },
+        {})
+  }
+
```

Comment:
Tests the successful uploading of multiple activity images, including compressing the images, uploading to Firebase Storage, deleting existing images, and updating the Firestore document with the new image URLs.